File: data-structures/non-linear/node/TreeNode.ts
class TreeNode {
  private value: number;
  private right: TreeNode | null;
  private left: TreeNode | null;

  constructor(
    value: number,
    right: TreeNode | null = null,
    left: TreeNode | null = null
  ) {
    this.value = value;
    this.right = right;
    this.left = left;
  }

  getValue(): number {
    return this.value;
  }

  getRight(): TreeNode | null {
    return this.right;
  }

  getLeft(): TreeNode | null {
    return this.left;
  }

  setValue(value: number): void {
    this.value = value;
  }

  setRight(node: TreeNode | null): void {
    this.right = node;
  }

  setLeft(node: TreeNode | null): void {
    this.left = node;
  }
}

export { TreeNode };

File: data-structures/non-linear/node/TreeNode.cpp
class TreeNode {
  int value;
  TreeNode *right;
  TreeNode *left;

public:
  TreeNode(int value, TreeNode *right = nullptr, TreeNode *left = nullptr)
      : value(value), right(right), left(left) {}

  int getValue() { return value; }

  TreeNode *getRight() { return right; }

  TreeNode *getLeft() { return left; }

  void setValue(int value) { this->value = value; }

  void setRight(TreeNode *node) { right = node; }

  void setLeft(TreeNode *node) { left = node; }
};
